platform/kobo/disable-wifi.sh

```diff
@@ -1,8 +1,15 @@
 #!/bin/sh
 
 # Disable wifi, and remove all modules.
-# NOTE: Trying to do this nicely with 'wpa_cli terminate' and 'dhcpcd -d -k "${INTERFACE}"' trips mysterious buggy corner-cases... (#6424)
-killall udhcpc default.script dhcpcd wpa_supplicant 2>/dev/null
+# NOTE: Save our resolv.conf to avoid ending up with an empty one, in case the DHCP client wipes it on release (#6424).
+cp -a "/etc/resolv.conf" "/tmp/resolv.ko"
+if [ -x "/sbin/dhcpcd" ]; then
+    env -u LD_LIBRARY_PATH dhcpcd -d -k "${INTERFACE}"
+else
+    killall udhcpc default.script 2>/dev/null
+fi
+mv -f "/tmp/resolv.ko" "/etc/resolv.conf"
```

Comment:
By principle (and to avoid ultra minor SDcard wear), not super fond of having stuff rewritten on the root partition everytime we'll disable Wifi. Can't we check if it's really needed ?

Reply:
I could over-engineer it and add a checksum, but, err, really? ^^. It's ~100 bytes.

Reply:
"principles" :) 
You could just `test -f` & `grep nameserver` and if boths are true, no need to, right?

I have this in my my_enable-wifi.sh:
```
    # On évite de recréer /etc/resolv.conf à chaque fois
    if ! grep -q '#MyStaticDNSconfig' /etc/resolv.conf 2>/dev/null  ; then
        echo '#MyStaticDNSconfig' > /etc/resolv.conf
        for i in $dns ; do echo nameserver $i >> /etc/resolv.conf ; done
    fi
```

Reply:
Ideally, I'd like to simply keep it as-is, so just checking if it's sane isn't quite enough.

Reply:
And, more to the point: it's already being rewritten by the DHCP client on acquire/renew/release, so I don't really see why we should be worried about *this* rewrite but not the others ;).

Reply:
(The fact that someone shitted in my living room doesn't mean that I should now too :)
But well, OK. (I'm not using DHCP, so I guess I'll have to patch all these scripts - but, I'm already doing that, so, I guess I'm OK.)

Reply:
> (I'm not using DHCP, so I guess I'll have to patch all these scripts - but, I'm already doing that, so, I guess I'm OK.)

That was kind of my point ;).

For 99.999% of the userbase, this is necessary ;).

Reply:
btw, are you sure the killall or dhcpd -k are done removing the resolv.conf when you are mv'ing it back?

Reply:
Yep, both are blocking!

Reply:
(Thanks for old_hash vs new_hash :)

Reply:
Was kind of free candy after the whole "let's implement killall -w manually" mess ^^.